extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```diff
@@ -0,0 +1,886 @@
+      const tempOutputTextFilePath = appendSuffixToFilePath(
+        outputTextFilePath,
+        i.toString(),
+      );
+      const tempFixDescriptionFilePath = appendSuffixToFilePath(
+        fixDescriptionFilePath,
+        i.toString(),
+      );
+      const tempTranscriptFilePath = appendSuffixToFilePath(
+        transcriptFilePath,
+        i.toString(),
+      );
+
+      tempOutputTextFiles.push(tempOutputTextFilePath);
+      fixDescriptionFiles.push(tempFixDescriptionFilePath);
+      transcriptFiles.push(tempTranscriptFilePath);
+
+      await runAutofixOnResults(
+        cocofixBin,
+        sarifFile,
+        srcRootPath,
+        tempOutputTextFilePath,
+        tempFixDescriptionFilePath,
+        tempTranscriptFilePath,
+        repoAutofixOutputStoragePath,
+        i,
+      );
+    }
+
+    // Merge the output files together.
+    // Caveat that autofix will call each alert "alert 0", which will look a bit odd in the merged output file.
+    await mergeFiles(tempOutputTextFiles, outputTextFilePath);
+    await mergeFiles(fixDescriptionFiles, fixDescriptionFilePath);
+    await mergeFiles(transcriptFiles, transcriptFilePath);
+  } else {
+    // Run autofix once for all alerts.
+    await runAutofixOnResults(
+      cocofixBin,
+      sarifFile,
+      srcRootPath,
+      outputTextFilePath,
+      fixDescriptionFilePath,
+      transcriptFilePath,
+      repoAutofixOutputStoragePath,
+    );
+  }
+
+  // Format the output text file with markdown.
+  await formatWithMarkdown(outputTextFilePath, `${nwo}`);
+
+  // Save output text files from each repo to later merge
+  // into a single markdown file containing all results.
+  outputTextFiles.push(outputTextFilePath);
+}
+
+/**
+ * Gets the storage paths for the autofix results for a given repository.
+ */
+async function getRepoStoragePaths(
+  autofixOutputStoragePath: string,
+  nwo: string,
+) {
+  // Create output directories for repo's autofix results.
+  const repoAutofixOutputStoragePath = join(
+    autofixOutputStoragePath,
+    nwo.replaceAll("/", "-"),
+  );
+  await ensureDir(repoAutofixOutputStoragePath);
+  return {
+    repoAutofixOutputStoragePath,
+    outputTextFilePath: join(repoAutofixOutputStoragePath, "output.txt"),
+    transcriptFilePath: join(repoAutofixOutputStoragePath, "transcript.md"),
+    fixDescriptionFilePath: join(
+      repoAutofixOutputStoragePath,
+      "fix-description.md",
+    ),
+  };
+}
+
+/**
+ * Runs autofix on the results in the given SARIF file.
+ */
+async function runAutofixOnResults(
+  cocofixBin: string,
+  sarifFile: string,
+  srcRootPath: string,
+  outputTextFilePath: string,
+  fixDescriptionFilePath: string,
+  transcriptFilePath: string,
+  workDir: string,
+  alertNumber?: number, // Optional parameter for specific alert
+): Promise<void> {
+  // Get autofix model from user settings.
+  const autofixModel = getAutofixModel();
+  if (!autofixModel) {
+    throw new Error("Autofix model not found.");
```

Comment:
Same here, would it be useful to add the name of the config setting?

Reply:
Done in https://github.com/github/vscode-codeql/commit/bf6d8e72b871f3068c1c89465d2ad527389d90de.